Developer Guide doc updates

chrisglein opened the pull request:
* Split "Code Style and Conventions" and "Source Code Structure" out of the Developer Guide into separate documents (unchanged). Linked to them from the top of the Developer Guide.
* Updated information on checks and pipelines
* Fix broken links in the documentation (where %20 was being used for deep links instead of hyphen)

jevansaks commented on docs/developer_guide.md:
```diff
@@ -119,6 +33,11 @@ an Administrator Powershell window:
 
 `.\build\Install-WindowsSdkISO.ps1 18312`
```
Can you update this to 18323? I forgot to in my last change.

jevansaks replied:
```suggestion
`.\build\Install-WindowsSdkISO.ps1 18323
```

licanhua commented on docs/source_code_structure.md:
```diff
@@ -0,0 +1,64 @@
+Our test library and test app (the app that the test library interacts with 
+when executing the tests) are here.
+
+MUXControls.Test is a [MSTest](https://docs.microsoft.com/dotnet/api/microsoft.visualstudio.testtools.unittesting) 
```
This supports both MSTest and TAEF test framework.

licanhua commented on docs/source_code_structure.md:
```diff
@@ -0,0 +1,64 @@
+when executing the tests) are here.
+
+MUXControls.Test is a [MSTest](https://docs.microsoft.com/dotnet/api/microsoft.visualstudio.testtools.unittesting) 
+DLL using MITALite that contains all of the test code for the various controls 
```
remove MITALite

chrisglein commented:
* Split "Code Style and Conventions" and "Source Code Structure" out of the Developer Guide into separate documents (unchanged). Linked to them from the top of the Developer Guide.
* Updated information on checks and pipelines
* Fix broken links in the documentation (where %20 was being used for deep links instead of hyphen)
